### PR TITLE
feat(can): api hardware sub project

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Adding a `hardware` sub project to the monorepo.

This will be the new home of the hardware controller. 

# Changelog

Create new `hardware` package adding Makefile, setup.py, etc.

# Review requests

# Risk assessment

None

### CODE REVIEW

File: hardware/hardware/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+"""Hardware package."""
```
Comment (mcous) [inline hardware/hardware/__init__.py]: Can we make this path `hardware/opentrons_hardware/__init__.py` to match the dist name?

File: hardware/mypy.ini [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+warn_unused_ignores = True
+disallow_untyped_decorators = True
+disallow_any_generics = True
+disallow_untyped_calls = True
+disallow_untyped_defs = True
+disallow_incomplete_defs = True
+no_implicit_optional = True
+warn_return_any = True
+no_implicit_reexport = True
+# end strict mode flags
```
Comment (mcous) [inline hardware/mypy.ini]:
```suggestion
strict = True
```

File: scripts/python_build_utils.py [hunk 3]
```diff
@@ -37,6 +37,9 @@
     'notify-server': PackageEntry(
         os.path.join(HERE, '..', 'notify-server', 'notify_server', 'package.json'),
         'notify_server'),
+    'hardware': PackageEntry(
+        os.path.join(HERE, '..', 'hardware', 'hardware', 'package.json'),
```
Comment (mcous) [inline scripts/python_build_utils.py]:
```suggestion
        os.path.join(HERE, '..', 'hardware', 'opentrons_hardware', 'package.json'),
```

**Review by mcous — APPROVED** [review]
🥳 
(Some little suggestions to avoid some future pain)

File: hardware/Pipfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+[packages]
+python-can = "==3.3.4"
+pyserial = "==3.5"
+typing-extensions = "==3.10.0.0"
+
+[dev-packages]
+pytest = "==6.1.0"
+pytest-aiohttp = "==0.3.0"
+pytest-cov = "==2.10.1"
+mypy = "==0.800"
```
Comment (mcous) [inline hardware/Pipfile]: Oh oops, can we use latest mypy to match `api` and `robot-server`? I think it's `0.910`

```suggestion
mypy = "==0.910"
```

File: hardware/Makefile [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+include ../scripts/push.mk
+include ../scripts/python.mk
+
+SHX := npx shx
+
+# Find the version of the wheel from package.json using a helper script. We
+# use python here so we can use the same version normalization that will be
+# used to create the wheel.
+wheel_file = dist/$(call python_get_wheelname,api,opentrons-hardware,$(BUILD_NUMBER))
```
Comment (sfoster1) [inline hardware/Makefile]: don't think `api` belongs here, not quite sure what does though

**Review by DerekMaggio — APPROVED** [review]
lgtm!

**Review by sfoster1 — APPROVED** [review]
Looks good!

**Pull request merged**